src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -62,6 +65,14 @@ public SqliteCommand(string commandText, SqliteConnection connection, SqliteTran
         }
 
         /// <summary>
+        /// Finalizes an instance of the <see cref="SqliteCommand"/> class.
+        /// </summary>
+        ~SqliteCommand()
+        {
+            Connection?.DisposePreparedStatements(_preparedStatements);
+        }
```

Comment:
You shouldn't need a finalizer. The base implementation calls `Dispose(false)`. Just remove the `if` block in the override below.